### PR TITLE
Introduce table upload convenience methods

### PR DESCRIPTION
**Pull request by xbrianh**
depends on #253 #254 #255

### CODE REVIEW

File: terra_notebook_utils/cli/table.py [hunk 1]
```diff
@@ -70,7 +71,7 @@ def fetch_drs_url(args: argparse.Namespace):
 
 @table_cli.command("put-row", arguments={
     "--table": dict(type=str, required=True, help="table name"),
-    "--id": dict(type=str, required=True, help="Row id. This should be unique for the table"),
+    "--row": dict(type=str, required=False, default=None, help="row name"),
```
Comment (DailyDreaming) [inline terra_notebook_utils/cli/table.py:74]: Do these still need to be unique?  If so, I thought that that was helpful info.  I might also document the behavior if this is not specified.

File: tests/test_table.py [hunk 2]
```diff
@@ -25,19 +25,8 @@ def test_get_access_token(self):
     @testmode("workspace_access")
     def test_table(self):
         table_name = f"test_{uuid4()}"
-        number_of_entities = 5
         table.delete(table_name)  # remove cruft from previous failed tests
```
Comment (DailyDreaming) [inline tests/test_table.py:28]: I don't understand this test.  Is this testing the deletion of a non-existent table?

**Review by DailyDreaming — APPROVED** [review]
See comments.

File: tests/test_table.py (hunk 2 above)
Comment (xbrianh) [reply tests/test_table.py:28]: These are updated in a separate PR.

File: terra_notebook_utils/cli/table.py (hunk 1 above)
Comment (xbrianh) [reply terra_notebook_utils/cli/table.py:74]: Fair point. Will update.

**Pull request merged**